### PR TITLE
Jumeg causality

### PR DESCRIPTION
**Pull request by dongqunxi**
@pravsripad @jdammers @fboers @lukasbreuer This is the initial scripts for ROIs definition and causality analysis.Thanks for all your help, hope in the future the strategy will become credible and stable.
Here are five documents:
dirs_manage: for building or reset derectory.
jumeg_causality: for parametric causality analysis.
MNE_ROIs_definition01: ROIs definition applying 'dSPM' or 'MNE'
ROIs display: Checking the common ROIs shown on the common brain.
test_MNE_ROIs_Definition: One example to show how to apply the ROIs functions.

### CODE REVIEW

File: jumeg_causality/dirs_manage.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import os
```
Comment (pravsripad) [inline jumeg_causality/dirs_manage.py]: @dongqunxi Can we put all of these functions into jumeg_utils.py file?

File: jumeg_causality/jumeg_causality.py [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+        p.append(np.linalg.lstsq(V.T, temp)[0].T)
+        P[i, :] = p[i][0] * np.ones((1, m))
+        for j in range(1, d1):
+            P[i, :] = x * P[i, :] + p[i][j]
+    Y = np.zeros(data.shape)
+    for r in xrange(N):
+        Y[:, :, r] = data[:, :, r] - P
+    return Y
+
+def normalize_data(fn_ts, factor=1):
```
Comment (pravsripad) [inline jumeg_causality/jumeg_causality.py]: @dongqunxi do these functions have a reference? e.g. from Scot?

Comment (dongqunxi) [inline jumeg_causality/jumeg_causality.py]: Hi Praveen,

These functions are mainly translated from matlab, I gave the links in
these scripts.

Best wishes,
Qunxi Dong

2015-08-25 16:25 GMT+02:00 Praveen Sripad <notifications@github.com>:

File: jumeg_causality/dirs_manage.py (hunk 1 above)
Comment (dongqunxi) [inline jumeg_causality/dirs_manage.py]: Hi Praveen,

Yes we can integrate and simplify it.

Best wishes,
Qunxi Dong

2015-08-25 16:25 GMT+02:00 Praveen Sripad <notifications@github.com>:

Comment (dongqunxi) [inline jumeg_causality/dirs_manage.py]: body { width: 100% !important; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; margin: 0; padding: 0; }
      img { outline: none; text-decoration: none; -ms-interpolation-mode: bicubic; }
      body { color: #65646a; font-family: 'Helvetica Neue', HelveticaNeue, helvetica, arial, sans-serif; }
      h1 a:active { color: #4a8db8 !important; }
      h2 a:active { color: #4a8db8 !important; }
      h3 a:active { color: #4a8db8 !important; }
      h4 a:active { color: #4a8db8 !important; }
      h5 a:active { color: #4a8db8 !important; }
      h6 a:active { color: #4a8db8 !important; }
      h1 a:visited { color: #4a8db8 !important; }
      h2 a:visited { color: #4a8db8 !important; }
      h3 a:visited { color: #4a8db8 !important; }
      h4 a:visited { color: #4a8db8 !important; }
      h5 a:visited { color: #4a8db8 !important; }
      h6 a:visited { color: #4a8db8 !important; }
      a:visited { color: #4a8db8; }
      a:hover { color: #2dbe60; text-decoration: underline; }
      .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
      .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
      .footer a:hover { text-decoration: none; }
      .footer-small a:hover { text-decoration: underline; }
      body { background-color: #f3f3f3; }
      .ExternalClass { width: 100%; }
      .ExternalClass { line-height: 100%; }
      table[id='inception'] .Btn-anchor:hover { color: #fff !important; }
      table[id='inception'] .Btn-anchor:visited { color: #fff !important; }
      @media screen and (min-width: 579px) {
        table[id="inception"] .l-inception-point { width: 578px !important; }
      }
      @media only screen and (min-device-width: 768px) and (max-device-width: 1024px) {
        table[id="inception"] a[href^="tel"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        table[id="inception"] a[href^="sms"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        .mobile-link table[id="inception"] a[href^="tel"] { text-decoration: default; color: #65646a !important; pointer-events: auto; cursor: default; }
        .mobile-link table[id="inception"] a[href^="sms"] { text-decoration: default; color: #65646a !important; pointer-events: auto; cursor: default; }
        table[id="inception"] .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
      }
      @media only screen and (max-device-width: 578px), only screen and (max-width: 578px) {
        table[id="inception"] .Branding_positive-container { padding: 14px 0 0 16px !important; }
        table[id='inception'] .Btn-anchor { display: block !important; }
        table[id='inception'] .Row_small { padding: 0 !important; }
        table[id='inception'] .Row_medium-top { padding-top: 16px !important; }
        table[id='inception'] .Avatar-img { width: 48px !important; height: 48px !important; }
        table[id="inception"] { width: 100% !important; }
        table[id="inception"] .l-outie { padding: 0px !important; }
        table[id="inception"] .l-inception-point { width: 100% !important; }
        table[id="inception"] .l-footer { width: 100% !important; }
        table[id="inception"] .l-wrap { width: 100% !important; }
        table[id="inception"] .l-full-bleed { width: 100% !important; }
        table[id="inception"] .media-obj { width: 100% !important; }
        table[id="inception"] .media-content { width: 100% !important; }
        table[id="inception"] .btn { width: 100% !important; }
        table[id="inception"] .btn td { width: 100% !important; }
        table[id="inception"] .l-inception-point { border-right: none !important; border-left: none !important; }
        table[id="inception"] .l-full-bleed img { width: 100% !important; height: auto !important; }
        table[id="inception"] .l-pad > tbody > tr > td { padding: 16px !important; }
        table[id="inception"] .l-split { width: 100% !important; }
        table[id="inception"] .l-split + .l-split { margin-top: 32px !important; }
        table[id="inception"] .ill-notebook { width: 100% !important; height: 96px !important; }
        table[id="inception"] .footer { padding-right: 16px !important; padding-left: 16px !important; }
        table[id="inception"] .btn span { display: block !important; }
        table[id="inception"] a[href^="tel"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        table[id="inception"] a[href^="sms"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        .mobile-link table[id="inception"] a[href^="tel"] { text-decoration: default; color: #4a8db8 !important; pointer-events: auto; cursor: default; }
        .mobile-link table[id="inception"] a[href^="sms"] { text-decoration: default; color: #4a8db8 !important; pointer-events: auto; cursor: default; }
        table[id="inception"] .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .Rule { display: none !important; }
        table[id="inception"] .l-pad { text-align: center !important; }
      }
    
  
  
    
      
        
          
            
              
                
                  
                    
                      
                        
                          
                          
                            
                              
                                
                                  
                                
                              
                            
                          
                        
                      
                    
                  
                  
                  
                    
                      
                        
                          
  
    
      

        
          
            
              

              
            

            
              
                
                  Dong Qunxi shared a note with you.
                
              
            

            
              
            
              View note in Evernote
        
              
            
          
        
      
    
  



  
    
      
        Write notes, collect ideas, and move your projects forward with Evernote.
      
    
  


                        
                      
                    
                  
                  
                
                
              
            
          
          
          
            
              
                For support requests, please contact us by going to our support page.

                This email has been sent to you by Evernote Corporation (or its subsidiary). Evernote Corporation, 305 Walnut Street, Redwood City, CA 94063, USA.

                
                  
                    Unsubscribe

Comment (dongqunxi) [inline jumeg_causality/dirs_manage.py]: body { width: 100% !important; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; margin: 0; padding: 0; }
      img { outline: none; text-decoration: none; -ms-interpolation-mode: bicubic; }
      body { color: #65646a; font-family: 'Helvetica Neue', HelveticaNeue, helvetica, arial, sans-serif; }
      h1 a:active { color: #4a8db8 !important; }
      h2 a:active { color: #4a8db8 !important; }
      h3 a:active { color: #4a8db8 !important; }
      h4 a:active { color: #4a8db8 !important; }
      h5 a:active { color: #4a8db8 !important; }
      h6 a:active { color: #4a8db8 !important; }
      h1 a:visited { color: #4a8db8 !important; }
      h2 a:visited { color: #4a8db8 !important; }
      h3 a:visited { color: #4a8db8 !important; }
      h4 a:visited { color: #4a8db8 !important; }
      h5 a:visited { color: #4a8db8 !important; }
      h6 a:visited { color: #4a8db8 !important; }
      a:visited { color: #4a8db8; }
      a:hover { color: #2dbe60; text-decoration: underline; }
      .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
      .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
      .footer a:hover { text-decoration: none; }
      .footer-small a:hover { text-decoration: underline; }
      body { background-color: #f3f3f3; }
      .ExternalClass { width: 100%; }
      .ExternalClass { line-height: 100%; }
      table[id='inception'] .Btn-anchor:hover { color: #fff !important; }
      table[id='inception'] .Btn-anchor:visited { color: #fff !important; }
      @media screen and (min-width: 579px) {
        table[id="inception"] .l-inception-point { width: 578px !important; }
      }
      @media only screen and (min-device-width: 768px) and (max-device-width: 1024px) {
        table[id="inception"] a[href^="tel"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        table[id="inception"] a[href^="sms"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        .mobile-link table[id="inception"] a[href^="tel"] { text-decoration: default; color: #65646a !important; pointer-events: auto; cursor: default; }
        .mobile-link table[id="inception"] a[href^="sms"] { text-decoration: default; color: #65646a !important; pointer-events: auto; cursor: default; }
        table[id="inception"] .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
      }
      @media only screen and (max-device-width: 578px), only screen and (max-width: 578px) {
        table[id="inception"] .Branding_positive-container { padding: 14px 0 0 16px !important; }
        table[id='inception'] .Btn-anchor { display: block !important; }
        table[id='inception'] .Row_small { padding: 0 !important; }
        table[id='inception'] .Row_medium-top { padding-top: 16px !important; }
        table[id='inception'] .Avatar-img { width: 48px !important; height: 48px !important; }
        table[id="inception"] { width: 100% !important; }
        table[id="inception"] .l-outie { padding: 0px !important; }
        table[id="inception"] .l-inception-point { width: 100% !important; }
        table[id="inception"] .l-footer { width: 100% !important; }
        table[id="inception"] .l-wrap { width: 100% !important; }
        table[id="inception"] .l-full-bleed { width: 100% !important; }
        table[id="inception"] .media-obj { width: 100% !important; }
        table[id="inception"] .media-content { width: 100% !important; }
        table[id="inception"] .btn { width: 100% !important; }
        table[id="inception"] .btn td { width: 100% !important; }
        table[id="inception"] .l-inception-point { border-right: none !important; border-left: none !important; }
        table[id="inception"] .l-full-bleed img { width: 100% !important; height: auto !important; }
        table[id="inception"] .l-pad > tbody > tr > td { padding: 16px !important; }
        table[id="inception"] .l-split { width: 100% !important; }
        table[id="inception"] .l-split + .l-split { margin-top: 32px !important; }
        table[id="inception"] .ill-notebook { width: 100% !important; height: 96px !important; }
        table[id="inception"] .footer { padding-right: 16px !important; padding-left: 16px !important; }
        table[id="inception"] .btn span { display: block !important; }
        table[id="inception"] a[href^="tel"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        table[id="inception"] a[href^="sms"] { text-decoration: none; color: #65646a; pointer-events: none; cursor: default; }
        .mobile-link table[id="inception"] a[href^="tel"] { text-decoration: default; color: #4a8db8 !important; pointer-events: auto; cursor: default; }
        .mobile-link table[id="inception"] a[href^="sms"] { text-decoration: default; color: #4a8db8 !important; pointer-events: auto; cursor: default; }
        table[id="inception"] .u-headingMobileAnchor a {text-decoration: none !important; color: #2dbe60 !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .u-bodyMobileAnchor a {text-decoration: none !important; color: #65646a !important; pointer-events: none !important; cursor: default !important;}
        table[id="inception"] .Rule { display: none !important; }
        table[id="inception"] .l-pad { text-align: center !important; }
      }
    
  
  
    
      
        
          
            
              
                
                  
                    
                      
                        
                          
                          
                            
                              
                                
                                  
                                
                              
                            
                          
                        
                      
                    
                  
                  
                  
                    
                      
                        
                          

  
    
      
        
          
            
              
                
                      You have a chat waiting in Evernote
                
              
            
            
              
                People want to chat with you. Click the green button to join in.
              
            
            
              
                
                      An unread chat from
                
              
            
            
              
                    Dong Qunxi
              
            
            
            
              
                Reply in Evernote
              
            
          
        
      
    
  



  
    
      
        Evernote is the one workspace you need. Collect ideas and work on projects together.
      
    
  


                        
                      
                    
                  
                  
                
                
              
            
          
          
          
            
              
                For support requests, please contact us by going to our support page.

                This email has been sent to you by Evernote Corporation (or its subsidiary). Evernote Corporation, 305 Walnut Street, Redwood City, CA 94063, USA.

                
                  
                    Unsubscribe

Comment (pravsripad) [inline jumeg_causality/dirs_manage.py]: @dongqunxi Hi Qunxi, (?) if this is some code copy pasted from Evernote, I think the formatting is messed up. Would you like to recheck?

Comment (dongqunxi) [inline jumeg_causality/dirs_manage.py]: @pravsripad, Yes I sent this information directly from Evernote, I thought it worked, we can delete them. @jdammers, can you also register one evernote account? I am checking the ICs related with ECG&EOG artifacts, and finishing reports of the preprocessing step on the evernote. You can check the report under the sharing notebook made by @pravsripad.

**Pull request closed without merging**